students/Dustin_L/lesson06/test_mailroom3.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python3
+"""This is a test module that tests mailroom3.py"""
+
+import os
+import sys
+import datetime
+import unittest
+from unittest import mock
+from io import StringIO
+from collections import defaultdict
+import mailroom3 as mr
+
+
+def redirect_stdout():
+    """Redirect stdout to the returned StringIO object.
+
+    Returns:
+        StringIO: Object containing redirected stdout.
+    """
+    redirect = StringIO()
+    sys.stdout = redirect
+
+    return redirect
+
+
+def reset_stdout():
+    """Reset stdout back to default"""
+    sys.stdout = sys.__stdout__
+
+
+class TestMailRoom(unittest.TestCase):
+
+    def setUp(self):
+        # Reset global donation database after each test case
+        mr.donor_db = defaultdict(lambda: mr.init_donor_data(),
+                                  {'Test Donor 1': mr.init_donor_data([1000, 5000, 10000]),
+                                   'Test Donor 2': mr.init_donor_data([12000, 5000, 27000]),
+                                   "Test Donor 3": mr.init_donor_data([38734, 6273, 67520])})
+
+    def tearDown(self):
+        # Remove any added thank you letters
+        files = os.listdir(os.getcwd())
+        for f in files:
+            if f.startswith('Test_Donor'):
+                os.remove(os.path.join(os.getcwd(), f))
+
+    def test_init_donor_data(self):
+        """Tests the init_donor_data() fxn"""
+
+        # Test with default parameter
+        ret_dict = mr.init_donor_data()
+        self.assertTrue(ret_dict[mr.GIFTS_KEY] == [],
+                        ret_dict[mr.GIFTS_KEY])
+        self.assertTrue(ret_dict[mr.NUM_GIFTS_KEY] == 0,
+                        ret_dict[mr.NUM_GIFTS_KEY])
+        self.assertTrue(ret_dict[mr.TOTAL_KEY] == 0,
+                        ret_dict[mr.TOTAL_KEY])
+        self.assertTrue(ret_dict[mr.AVE_KEY] == 0,
+                        ret_dict[mr.AVE_KEY])
+
+        # Test with argument
+        gifts = [5, 10, 20, 15]
+        ret_dict = mr.init_donor_data(gifts=gifts)
+        self.assertTrue(ret_dict[mr.GIFTS_KEY] == gifts,
+                        ret_dict[mr.GIFTS_KEY])
+        self.assertTrue(ret_dict[mr.NUM_GIFTS_KEY] == 4,
+                        ret_dict[mr.NUM_GIFTS_KEY])
+        self.assertTrue(ret_dict[mr.TOTAL_KEY] == 50,
+                        ret_dict[mr.TOTAL_KEY])
+        self.assertTrue(ret_dict[mr.AVE_KEY] == 12.5,
+                        ret_dict[mr.AVE_KEY])
```

Comment:
this would normally go into a second test -> `def test_init_donor_data_with_gifts`